testssl.sh

```diff
@@ -10909,6 +10947,41 @@ run_fs() {
                     [[ $i -eq $nr_curves ]] && break
                     supported_curve[i]=true
                done
+               # Versions of TLS prior to 1.3 close the connection if the client does not support the curve
+               # used in the certificate. The easiest solution is to move the curves to the end of the list.
+               # instead of removing them from the ClientHello. This is only needed if there is no RSA certificate.
+               while true; do
+                    if ([[ "$proto" == 03 ]] && [[ ! "$ecdhe_cipher_list" == *RSA* ]]) || break; then
+                         curves_to_test=""
+                         for (( i=0; i < nr_curves; i++ )); do
+                              if ! "${curves_deprecated[i]}" || [[ "$proto" == 03 ]]; then
```

Comment:
This "if" statement should be deleted. As noted above, deprecated curves should only be excluded from TLS 1.3 ClientHello messages. Also, there is no need to check that `"$proto$ == 03`, as that is done earlier.